### PR TITLE
bugfix: str type var builtin function hover

### LINKED ISSUE #758: [Bug] String variable missing member function hover
a = "".capitalize() # The function has hover info
b = a.capitalize() # The function has not hover info

### PR DESCRIPTION
**Pull request by He1pa**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y  fix #758  

#### 2. What is the scope of this PR (e.g. component or file name):
tools/src

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):
bugfix: str type var builtin function hover

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: kclvm/tools/src/LSP/src/goto_def.rs [hunk 1]
```diff
@@ -334,6 +334,30 @@ pub(crate) fn resolve_var(
                             None => None,
                         }
                     }
+                    kclvm_sema::ty::TypeKind::Str => {
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/goto_def.rs:337]:
```rust
kclvm_sema::ty::TypeKind::Str if names.len() == 2 {
}
```

File: kclvm/tools/src/LSP/src/goto_def.rs [hunk 2]
```diff
@@ -334,6 +334,30 @@ pub(crate) fn resolve_var(
...
+                        if names.len() == 2 {
+                            let func_name_node = node_names[1].clone();
+                            let func_name = func_name_node.node.clone();
+                            match STRING_MEMBER_FUNCTIONS.get(&func_name) {
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/goto_def.rs:341]: if let Some(..) = xx

**Review by i-zhen — APPROVED** [review]
LGTM

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**